### DISCUSSION TITLE
Welcome to ReVanced Patches Discussions!

### DISCUSSION THREAD

**Discussion by anddea** [anddea/revanced-patches#128] [Announcements]
## 👋 Welcome!
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and be open-minded. Remember that this is a community we  build together 💪.

**Comment by hariharan-cmuthu** [comment]
Hi there, could we expect a fix to the custom branding patch on reddit? Cause whenever i try to login (with my username & password, not by google or other means!) to reddit with that specific patch there's this error message "something went wrong" I've tried with multiple apks but the issue still exists. Thanks.

**Reply by anddea** [reply]
I'm not sure how custom branding can be related to logging in, but I'll take a look.

**Comment by Aryan-Chauhan** [comment]
Can we get support for hiding more texts?

https://github.com/anddea/revanced-patches/commit/d8b1da2180bb9ffbce2d8febd635da41ce06aa36

**Reply by anddea** [reply]
> Can we get support for hiding more texts?

Like what?

**Reply by Aryan-Chauhan** [reply]
> > Can we get support for hiding more texts?
> 
> Like what?

I was referring to these settings (completely useless).

![Screenshot_2024-04-01-05-19-12-28_f22faf49515899f9781ec174a4d85388.jpg](https://github.com/anddea/revanced-patches/assets/91815107/23c37f93-7d74-4e25-8ded-f32346c90416)

**Reply by nullll1** [reply]
On my side, I have more useless settings

![Screenshot_20240331-182314~2.png](https://github.com/anddea/revanced-patches/assets/102777708/ed2fd2e4-19ca-4dac-8c99-3abedc18e53c)

these settings r useless for me

**Reply by Aryan-Chauhan** [reply]
> On my side, I have more useless settings
> 
> ![Screenshot_20240331-182314~2.png](https://github.com/anddea/revanced-patches/assets/102777708/ed2fd2e4-19ca-4dac-8c99-3abedc18e53c)
> 
> these settings r useless for me
> 
> 

Bruh, I also had those same settings like you, but I hid them using anddea's latest patch. Now, I'm requesting for the ability to hide remaining ones.

Write this exactly in patch options of "Hide Settings Preferences" to make your settings same as my screenshot:

![Screenshot_2024-04-01-06-03-40-76_589a91589cdcdbd1b87a5806ff4a71ba.jpg](https://github.com/anddea/revanced-patches/assets/91815107/e2cf14ca-b2b3-4558-a4ef-1d6fee5de579)

**Reply by nullll1** [reply]
I'm also using anddea's latest dev patch, I don't see the patch "hide setting preferences"

**Reply by Aryan-Chauhan** [reply]
> I'm also using anddea's latest dev patch, I don't see the patch "hide setting preferences"

Due to some unknown reason, it is not showing on RVX Manager or ReVanced Manager.

I use ReVancify.

**Reply by anddea** [reply]
I have added more options, please check and let me know because I don't have these entries in the settings.

"Account" and "Your data in YouTube" won't be hidden, because hiding them causes a crash. When I have more time and patience, I'll look for possible ways to fix that.

**Reply by Aryan-Chauhan** [reply]
Thanks, anddea, for quickly implementing so many cool features!

Everything is hiding fine, except Purchases and memberships.

Also, it should be required by the user to be written as "Billing and payments", and not "Billing & payments" because 'and' is used in unpatched YT app, not '&'.

![Screenshot_2024-04-01-20-02-12-10_07f20907771ad09ffa2dcbecf65bb740.jpg](https://github.com/anddea/revanced-patches/assets/91815107/532bf3ed-788b-459e-9dc8-348329ef5d16)

![Screenshot_2024-04-01-20-23-29-55_f22faf49515899f9781ec174a4d85388.jpg](https://github.com/anddea/revanced-patches/assets/91815107/c566386f-ef4d-4302-84bf-d5fbaff14d2b)

**Reply by nullll1** [reply]
I'm using rvx manager and the setting "Billing  & payments" is hiding and I don't need to change the & to and


![Screenshot_20240401-090337.png](https://github.com/anddea/revanced-patches/assets/102777708/8848a5db-eeed-4340-babc-e1fc9077648b)

![Screenshot_20240401-090724.png](https://github.com/anddea/revanced-patches/assets/102777708/a9b9af6c-3e53-438d-97df-83839ed7d70d)

**Reply by Aryan-Chauhan** [reply]
> I'm using rvx manager and the setting "Billing  & payments" is hiding and I don't need to change the & to and
> 
> 
> ![Screenshot_20240401-090337.png](https://github.com/anddea/revanced-patches/assets/102777708/8848a5db-eeed-4340-babc-e1fc9077648b)
> 
> ![Screenshot_20240401-090724.png](https://github.com/anddea/revanced-patches/assets/102777708/a9b9af6c-3e53-438d-97df-83839ed7d70d)
> 
> 

I never said, you need to change "&" to "and".

I said, it should be "and" in future patch releases.

**Reply by anddea** [reply]
I guess, it varies, it is Billing & payments in this screenshot https://github.com/anddea/revanced-patches/discussions/128#discussioncomment-8967947. So I added both spellings, they should hide the same option. Now "Purchases and memberships" should be hidden. Can you verify and let me know?

**Reply by Aryan-Chauhan** [reply]
> I guess, it varies, it is Billing & payments in this screenshot https://github.com/anddea/revanced-patches/discussions/128#discussioncomment-8967947. So I added both spellings, they should hide the same option. Now "Purchases and memberships" should be hidden. Can you verify and let me know?

Yes, "Purchases and memberships" can be hidden now. Quick fix!

**Reply by nullll1** [reply]
yes, it's hidden, what a quick fix

**Reply by sultanmonsur** [reply]
Revancify not working 😞 
Revancify showing limited options.
![IMG_20240402_111440](https://github.com/anddea/revanced-patches/assets/61897416/daaa6d39-fab1-4277-9e41-b2c4ec78add7)

**Reply by Aryan-Chauhan** [reply]
> Revancify not working 😞 
> Revancify showing limited options.
> ![IMG_20240402_111440](https://github.com/anddea/revanced-patches/assets/61897416/daaa6d39-fab1-4277-9e41-b2c4ec78add7)
> 

Go to your phone's internal storage, then ReVancify folder, then open "anddea-options.json" and type exactly like this:
![Screenshot_2024-04-02-14-04-23-78_07f20907771ad09ffa2dcbecf65bb740.jpg](https://github.com/anddea/revanced-patches/assets/91815107/a8ee1e0c-a5c0-4f1a-bce0-66de034b440f)

**Reply by Francesco146** [reply]
@sultanmonsur it should be fixed with the latest commit made by Revancify. use this:
`revancify -r`

**Comment by Fivesal** [comment]
Patch error


Error occurred during initialization of VM
Unable to load native library: dlopen failed: library "libandroid-spawn.so" not found



Rooted: false
Arch: arm64-v8a
App: YouTube v19.11.38
CLI: inotia00-cli-v4.4.1.jar
Patches: anddea-patches-v2.223.0.jar
Integrations: anddea-integrations-v0.150.0.apk
Patches argument: -i Add splash animation -i Alternative thumbnails -i Ambient mode switch -i Append time stamps information -i Change start page -i Custom double tap length -i Custom package name -i Custom playback speed -i Custom player overlay opacity -i Custom seekbar color -i Default playback speed -i Default video quality -i Disable HDR video -i Disable QUIC protocol -i Disable auto captions -i Disable haptic feedback -i Disable landscape mode -i Disable pip notification -i Disable rolling number animations -i Disable shorts on startup -i Disable speed overlay -i Disable update screen -i Enable bottom player gestures -i Enable compact controls overlay -i Enable external browser -i Enable gradient loading screen -i Enable language switch -i Enable minimized playback -i Enable new splash animation -i Enable new thumbnail preview -i Enable old quality layout -i Enable open links directly -i Enable seekbar tapping -i Enable song search -i Enable tablet mini player -i Enable tablet navigation bar -i Enable wide search bar -i Force fullscreen -i Force opus codec -i Force video codec -i Header switch -i Hide account menu -i Hide auto player popup panels -i Hide autoplay button -i Hide autoplay preview -i Hide button container -i Hide captions button -i Hide cast button -i Hide category bar -i Hide channel avatar section -i Hide channel profile components -i Hide channel watermark -i Hide collapse button -i Hide comment component -i Hide crowdfunding box -i Hide description components -i Hide double tap overlay filter -i Hide end screen cards -i Hide end screen overlay -i Hide feed flyout panel -i Hide filmstrip overlay -i Hide floating microphone -i Hide fullscreen panels -i Hide general ads -i Hide handle -i Hide info cards -i Hide latest videos button -i Hide layout components -i Hide load more button -i Hide mix playlists -i Hide music button -i Hide navigation buttons -i Hide navigation label -i Hide player flyout panel -i Hide previous next button -i Hide search term thumbnail -i Hide seek message -i Hide seekbar -i Hide shorts components -i Hide snack bar -i Hide suggested actions -i Hide suggested video overlay -i Hide suggestions shelf -i Hide time stamp -i Hide toolbar button -i Hide tooltip content -i Hide trending searches -i Hide video ads -i Keep landscape mode -i Layout switch -i MicroG support -i Overlay buttons -i Quick actions components -i Remove viewer discretion dialog -i Return YouTube Dislike -i Sanitize sharing links -i Settings -i Shorts overlay buttons -i SponsorBlock -i Spoof app version -i Spoof device dimensions -i Swipe controls -i Theme -i Translations -i Tuck away preferences -i Visual preferences icons

**Reply by anddea** [reply]
It's a Termux problem. If you fixed it, let everyone know how you did that, so they can manage it too when they face this issue.

**Reply by aqtoo** [reply]
I opened an issue on this and was given a solution, for some reason some packages just don't seem to install during the downloads of said packages.

SOLUTION HERE : https://github.com/decipher3114/Revancify/issues/72#issuecomment-2032052006

**Comment by Raihan-Shovo** [comment]
I'm not sure if this happened to anybody but when I'm using YouTube after few minutes or few videos or even sometimes one video or some short, the YouTube suddenly crashes and it's happening like every single time when using YouTube. I don't know that it's just me or somebody else have the same issue. Don't know that is it some kind of a bug or something else. I am using an old Android and it's running on Android 8 and YT has the latest patch.

**Reply by Aryan-Chauhan** [reply]
> I'm not sure if this happened to anybody but when I'm using YouTube after few minutes or few videos or even sometimes one video or some short, the YouTube suddenly crashes and it's happening like every single time when using YouTube. I don't know that it's just me or somebody else have the same issue. Don't know that is it some kind of a bug or something else. I am using an old Android and it's running on Android 8 and YT has the latest patch.
> 

Can you tell me these details:

Phone name:

OS:

YT version:

Patches version:

Integrations version:

GmsCore you are using and it's version (tell repo name also where you downloaded it from):

**Reply by Raihan-Shovo** [reply]
> > I'm not sure if this happened to anybody but when I'm using YouTube after few minutes or few videos or even sometimes one video or some short, the YouTube suddenly crashes and it's happening like every single time when using YouTube. I don't know that it's just me or somebody else have the same issue. Don't know that is it some kind of a bug or something else. I am using an old Android and it's running on Android 8 and YT has the latest patch.
> > 
> 
> Can you tell me these details:

Phone name: Vivo Y91i

OS: Android 8.0

YT version: 19.11.38

Patches version: 2.223.0

Integrations version: 0.150.0

GmsCore you are using and it's version (tell repo name also where you downloaded it from): MicroG-RE V5.3 https://github.com/WSTxda/MicroG-RE

**Reply by Aryan-Chauhan** [reply]
> > > I'm not sure if this happened to anybody but when I'm using YouTube after few minutes or few videos or even sometimes one video or some short, the YouTube suddenly crashes and it's happening like every single time when using YouTube. I don't know that it's just me or somebody else have the same issue. Don't know that is it some kind of a bug or something else. I am using an old Android and it's running on Android 8 and YT has the latest patch.
> > > 
> > 
> > Can you tell me these details:
> 
> Phone name: Vivo Y91i
> 
> OS: Android 8.0
> 
> YT version: 19.11.38
> 
> Patches version: 2.223.0
> 
> Integrations version: 0.150.0
> 
> GmsCore you are using and it's version (tell repo name also where you downloaded it from): MicroG-RE V5.3 https://github.com/WSTxda/MicroG-RE
> 
> 

Clean-install v19.05.36 with latest patches & integrations. Maybe your phone has some problem with new YT app versions.

BTW, you have no benefit using v19.11.38 because it doesn't have one single new thing compared to v19.05.36. Only the number looks bigger.

If you use v19.05.36, you will miss out NOTHING.

Also, the microG you are using is old & has playback issues, so use this:

https://github.com/ReVanced/GmsCore

Or this (both are same, this one has low APK size):

https://github.com/YT-Advanced/GmsCore

**Reply by Raihan-Shovo** [reply]
> > > > I'm not sure if this happened to anybody but when I'm using YouTube after few minutes or few videos or even sometimes one video or some short, the YouTube suddenly crashes and it's happening like every single time when using YouTube. I don't know that it's just me or somebody else have the same issue. Don't know that is it some kind of a bug or something else. I am using an old Android and it's running on Android 8 and YT has the latest patch.
> > > > 
> > > 
> > > Can you tell me these details:
> > 
> > Phone name: Vivo Y91i
> > 
> > OS: Android 8.0
> > 
> > YT version: 19.11.38
> > 
> > Patches version: 2.223.0
> > 
> > Integrations version: 0.150.0
> > 
> > GmsCore you are using and it's version (tell repo name also where you downloaded it from): MicroG-RE V5.3 https://github.com/WSTxda/MicroG-RE
> > 
> > 
> 
> Clean-install v19.05.36 with latest patches & integrations. Maybe your phone has some problem with new YT app versions.
> 
> BTW, you have no benefit using v19.11.38 because it doesn't have one single new thing compared to v19.05.36. Only the number looks bigger.
> 
> If you use v19.05.36, you will miss out NOTHING.
> 
> Also, the microG you are using is old & has playback issues, so use this:
> 
> https://github.com/ReVanced/GmsCore
> 
> Or this (both are same, this one has low APK size):
> 
> https://github.com/YT-Advanced/GmsCore

I have tried all GmsCore and have the same issue. I will try to downgrade the app version, I will let you know if anything happens.
Edit: @Aryan-Chauhan downgraded the YT and it worked, no more crashes. Thanks.

**Comment by Francesco146** [comment]
@anddea regarding your last commit, can you give us some documentation about the `parent settings` in the `Tuck away preferences` patch? I'm referring to this commit: 137d680

**Reply by anddea** [reply]
It appears only for people who has enabled "Parent controls" aka "Parent settings". It goes on top of General.

**Comment by Francesco146** [comment]
@anddea hi! sorry to bother you. i'm learning how to create a simple Litho patch. i can't understand how can i retrieve a component name from the logs. the whole goal is learning how to do this stuff, and create a patch the hides the comment button if the comments on that Short are disabled or 0.

i've reached this point:
```bash
shorts_action_bar.eml|hash|ContainerType(x3)|shorts_video_action_button.eml|hash|ContainerType|

shorts_action_bar.eml|hash|ContainerType(x3)|shorts_video_action_button.eml|hash|ContainerType(x2)|

shorts_action_bar.eml|hash|ContainerType(x3)|shorts_video_action_button.eml|hash|ContainerType(x3)|

shorts_action_bar.eml|hash|ContainerType(x3)|shorts_video_action_button.eml|hash|ContainerType(x3)|TextType|

shorts_action_bar.eml|hash|ContainerType(x3)|shorts_video_action_button.eml|hash|ContainerType(x3)|ImageType|
```

i think i'm on the right track but i don't understand how can i pull the component name out of this. honestly, since the hash it's generated at runtime, i'm not sure this can even be done with a simple custom filter like `short_comment_button$0` or `short_comment_button$Disabled`. what do you think? ty :D

**Reply by anddea** [reply]
Hi. Litho paths you sent do not show any information about disabled or 0, probably you'll have to check buffer logs as well and if it has anything related to that you can filter with buffer filtering. And if there's no such information, you cannot do this with litho filter.

**Reply by anddea** [reply]
I checked. There's a number of comments in buffer and no number means disabled (I guess), you can use regex to grab that. Although, hiding comments button for 0 can be unwanted, maybe someone would want to leave a comment.

**Reply by Francesco146** [reply]
> you can use regex to grab that. 

`([a-zA-Z]+)` this should do the trick. but this can't be applied through the Custom filters patch like i imagined, i need to create a Kotlin patch, right? Do you happen to have a similar patch so I can look into what to do? sry but it's my first time touching anything like this :_) and I don't know where to look

> Although, hiding comments button for 0 can be unwanted, maybe someone would want to leave a comment.

sure, good point

**Reply by anddea** [reply]
The regex you sent grabs every word but you need number only smth like \d+, and if there's no number for such element then consider it as disabled and hide it. 

There's nothing similar, but you can search for "matcher" to see how regex is used in different parts. 

Here's how comments button is being grabbed (and later hidden) 
https://github.com/anddea/revanced-integrations/blob/6e75885d51e4f0df554830545382dca587ed0925/app/src/main/java/app/revanced/integrations/youtube/patches/components/ShortsFilter.java#L136-L141

It already grabs buffer, you have to do the same but with new setting and then in filter logic (in the same file) you will have to use pattern matching, smth like

PATTERN = "\d+"

if (NEW_SETTING && PATTERN.matcher(protobufBufferArray).find()) return false

Which means do not filter if pattern with any numbers is found, it means it's not disabled. But it's just an example, in reality it can be different, you have to test it.

**Reply by Francesco146** [reply]
thank you for the infos, quite complex doing this the first time! I'll take a look at this in the weekend :)

**Comment by Morafoka69** [comment]
Please how do I know the exact latest recommended version for YT Music? In your repo it says 6.21.52+.

Thank you in anticipation.

**Reply by anddea** [reply]
Open apkmirror and download latest YT Music. It supports all version after 6.21, unless something will be broken.

**Reply by Morafoka69** [reply]
Thank you so very mụ for replying. Does this apply to the armeabi-v7a version?

**Reply by anddea** [reply]
Yes, any architecture but it should apk. It might depend on what patching tool are you using. From what I know they should all support all architectures.

**Comment by Morafoka69** [comment]
Please I saw Reddit in your latest patch release. Can Reddit now be patched with RVX Manager?

Thanks.

**Reply by Francesco146** [reply]
no

**Reply by anddea** [reply]
For new versions only CLI

**Comment by Zonkerus** [comment]
Will you no longer be releasing new patches?

![image](https://github.com/anddea/revanced-patches/assets/166645857/e38862c8-a1fb-4f33-83bd-0f53042873e3)

**Reply by anddea** [reply]
I will. Not all changes will be merged and if people need them they can use this repo, but for most, there is no need to use these patches anymore.

**Reply by BAXSD** [reply]
> I will. Not all changes will be merged and if people need them they can use this repo, but for most, there is no need to use these patches anymore.

Yes stay :) I really love your work and patches. Have gotten used to it and dont wanna switch to another :)

**Comment by Morafoka69** [comment]
Please how I stop this from happening?

This is RVX Reddit with Anddea patches.

Thanks in anticipation. 
![Screenshot_20240614-035107](https://github.com/anddea/revanced-patches/assets/155040570/f81ab2b7-860e-402f-a157-5daec19603e2)

**Reply by anddea** [reply]
I don't know, maybe other community members can help you

**Reply by Francesco146** [reply]
which version have you patched?

**Reply by Morafoka69** [reply]
The latest version of Reddit. It has been happening before CLI was the only option of patching it.

**Comment by Morafoka69** [comment]
In this release, you said that you added MMT Blue, MMT Green and MMT Yellow icons:

https://github.com/anddea/revanced-patches/releases/tag/v2.226.0-dev.15

But, I checked the RVX Manager and attached is a screenshot showing what I found. I'm using your latest patches.

Also, do you have a Discord or Telegram Channel? Thanks in anticipation. 
![Screenshot_20240622-011505](https://github.com/anddea/revanced-patches/assets/155040570/61b63255-96c2-46ed-b1df-f789da17cdfc)

**Reply by Aryan-Chauhan** [reply]
Are you using pre-releases in Manager or not?

**Reply by anddea** [reply]
If you enabled pre-releases but still don't see it then just clear cache or data of the Manager. It can be done inside Manager by clearing options, temporary files, etc.

**Reply by Morafoka69** [reply]
> Are you using pre-releases in Manager or not?

Yes I am.

**Reply by Morafoka69** [reply]
> If you enabled pre-releases but still don't see it then just clear cache or data of the Manager. It can be done inside Manager by clearing options, temporary files, etc.

Thanks for the info. Let me try it.

**Comment by Morafoka69** [comment]
Error patching YT.

I'm patching the latest recommended version of YT and everything goes smooth until it starts packing the dex files. Please help. I'm attaching a screenshot and a log file. I'm waiting for your reply. 

[YT.logs.txt](https://github.com/user-attachments/files/16041912/YT.logs.txt)
![Screenshot_20240629-200544](https://github.com/anddea/revanced-patches/assets/155040570/02fdaa74-c45a-40ad-9413-f678439017eb)
Thank you in anticipation.

**Reply by Francesco146** [reply]
clean the manager cache and data, after this retry

**Comment by Aurora3rs** [comment]
Patch Error
- using recommended yt version (19.16.39)
- using latest RVX manager (1.20.4)
- things already done: cleared RVX manager cache and data & reset patch options & deleted temporary files

Idk if I'm doing anything wrong like selection a patch option I shouldn't so correct me.
Thanks in advance.

![Screenshot_20240725_193822.jpg](https://github.com/user-attachments/assets/3beb3d8f-e5eb-4055-aaca-e3a95cf9c1b8)


![RVX - Patch Log.txt](https://github.com/user-attachments/files/16380643/RVX.-.Patch.Log.txt)

**Reply by anddea** [reply]
This is caused by custom branding icon, YouTube icon and Vanced icons too. If you disable this patch at all it'll use YouTube icon anyway. I'll try to fix it in 24 hours. Thank you for your report.

**Reply by Aurora3rs** [reply]
Okay. Thank you.

**Reply by anddea** [reply]
Should be fixed now in dev release

**Reply by Aurora3rs** [reply]
It's fixed. Thanks G.

**Comment by ghost** [comment]
> ## 👋 Welcome!
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and be open-minded. Remember that this is a community we  build together 💪.

**Comment by ghost** [comment]
Hello anddea. Thank you for working on the best project on Github, I use RVX everyday and I am grateful that I can watch YT without ads. I am not a Dev but I am really interested Tech in general. I would like to work on your project, contact me if you are interested.

**Reply by anddea** [reply]
Hello!

We welcome any contributions you can make to our project. We have a diverse team of contributors, including coders, reverse engineers, resource managers, graphic designers, icon creators, translators, proofreaders, social media managers, and more.

To get started, you can join [Telegram channel](https://t.me/AnddeaChat) for this repo only. There, you'll find some contributors, not only for this project but also for related projects like the original ReVanced, inotia's, ReX's, Francesco's projects, and many others. We all collaborate to make this project a success, and it's not just a one-person effort.

If you want to contribute to the source code, it would be great if you have experience in any of the areas mentioned above (coding, reverse engineering, designing, etc.). Additionally, it's very important to have the technical skills to work with source code, such as building, compiling, patching, and having a good understanding of the command line and other basic tools.

Check this documentation below for basics:
https://github.com/ReVanced/revanced-documentation

**Reply by Morafoka69** [reply]
> Francesco's projects

Please what are these? Can you explain briefly? TIA.

**Reply by anddea** [reply]
Francesco is the contributor and the part-time maintainer of this repo. 

https://github.com/Francesco146/revanced-patches
https://github.com/Francesco146/revanced-integrations

His guide for beginners, still working on it from time to time
https://github.com/Francesco146/revanced-for-dummies

**Comment by Morafoka69** [comment]
Please how do I hide this? Thanks. 
![Screenshot_20241110-231223_1](https://github.com/user-attachments/assets/96f3ee6f-a5b7-45ec-8b3b-5ba96680181f)
